src/hotspot/cpu/x86/c2_MacroAssembler_x86.cpp

```diff
@@ -811,9 +811,16 @@ void C2_MacroAssembler::fast_unlock(Register objReg, Register boxReg, Register t
 #endif
     jccb(Assembler::zero, Stacked);
     if (UseFastLocking) {
-      // If the owner is ANONYMOUS, we need to fix it - in the slow-path.
+      // If the owner is ANONYMOUS, we need to fix it.
       testb(Address(tmpReg, OM_OFFSET_NO_MONITOR_VALUE_TAG(owner)), (int32_t) (intptr_t) ANONYMOUS_OWNER);
+#ifdef _LP64
+      C2FixAnonOMOwnerStub* stub = new (Compile::current()->comp_arena()) C2FixAnonOMOwnerStub(tmpReg);
+      Compile::current()->output()->add_stub(stub);
+      jcc(Assembler::notEqual, stub->entry());
+      bind(stub->continuation());
+#else
       jcc(Assembler::notEqual, NO_COUNT);
```

Comment:
Could you add a comment "// not yet implemented" ? Just to show that there is nothing basic stopping us from doing this for 32-bit, its just not yet done.

Reply:
I'll just implement it. *shrugs*

Reply:
In-fact, this is not so easy. See the comment that I added.